lib/dependencies/ImportMetaPlugin.js

```diff
@@ -0,0 +1,77 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Ivan Kopeykin @vankop
+*/
+
+"use strict";
+
+const {
+	toConstantDependency
+} = require("../javascript/JavascriptParserHelpers");
+const ImportMetaDependency = require("./ImportMetaDependency");
+
+/** @typedef {import("../Compiler")} Compiler */
+/** @typedef {import("../javascript/JavascriptParser")} Parser */
+
+class ImportMetaPlugin {
+	/**
+	 * @param {Compiler} compiler compiler
+	 */
+	apply(compiler) {
+		compiler.hooks.compilation.tap(
+			"ImportMetaPlugin",
+			(compilation, { normalModuleFactory }) => {
+				compilation.dependencyTemplates.set(
+					ImportMetaDependency,
+					new ImportMetaDependency.Template()
+				);
+				/**
+				 * @param {Parser} parser parser
+				 * @param {Object} parserOptions parserOptions
+				 * @returns {void}
+				 */
+				const parserHandler = (parser, parserOptions) => {
+					parser.hooks.typeof
+						.for("import.meta")
+						.tap(
+							"ImportMetaPlugin",
+							toConstantDependency(parser, JSON.stringify("object"))
+						);
+					parser.hooks.typeof
+						.for("import.meta.url")
+						.tap(
+							"ImportMetaPlugin",
+							toConstantDependency(parser, JSON.stringify("string"))
+						);
+
+					parser.hooks.metaProperty.tap("ImportMetaPlugin", expr => {
+						parser.state.module.addPresentationalDependency(
```

Comment:
We probably add more properties to `import.meta` in future and this would probably result in using `import.meta` without property access getting pretty inconsistent is usage. Especially when we add methods to `import.meta`.

I would say that using `import.meta` directly should be disallowed -> resulting in a CriticalDependency warning, just like using `require` without calling it. So using `import.meta` should result in `{}` + warning

Handling this without warning could lead to the false impression that something like that is supported:

``` js
// wrong:
const { resolve } = import.meta;
resolve("./module");

// correct:
import.meta.resolve("./module");
```

Reply:
`evaluating` `import.meta` should be added. Something with `new BasicEvaluatedExpression().setIdentifier().setTruthy().setRange()`